src/rust/engine/grpc_util/src/tls.rs

```diff
@@ -8,18 +8,36 @@ use tokio_rustls::rustls::{Certificate, ClientConfig, Error, RootCertStore, Serv
 
 #[derive(Default, Clone)]
 pub struct Config {
-  pub root_ca_certs: Option<Vec<u8>>,
+  pub root_ca_certs: Option<Vec<Certificate>>,
   pub mtls: Option<MtlsConfig>,
   pub certificate_check: CertificateCheck,
 }
 
 impl Config {
-  pub fn new_without_mtls(root_ca_certs: Option<Vec<u8>>) -> Self {
-    Self {
+  /// Creates a new config with the given root CA certs and mTLS config.
+  pub fn new(root_ca_certs: Option<&[u8]>, mtls: MtlsConfig) -> Result<Self, String> {
```

Comment:
Maybe just merge the two methods into a `new` which takes `root_ca_certs` and `mtls` both as options?